R/theme.r

```diff
@@ -557,23 +511,31 @@ calc_element <- function(element, theme, verbose = FALSE) {
 
   # If no parents, this is a "root" node. Just return this element.
   if (is.null(pnames)) {
+    if (verbose) message("nothing (top level)")
+
     # Check that all the properties of this element are non-NULL
-    nullprops <- vapply(theme[[element]], is.null, logical(1))
-    if (any(nullprops)) {
-      stop("Theme element '", element, "' has NULL property: ",
-        paste(names(nullprops)[nullprops], collapse = ", "))
+    nullprops <- vapply(el_out, is.null, logical(1))
+    if (!any(nullprops)) {
+      return(el_out) # no null properties, return element as is
     }
 
-    if (verbose) message("nothing (top level)")
-    return(theme[[element]])
+    # if we have null properties, try to fill in from theme_gray()
+    el_out <- combine_elements(el_out, theme_gray()[[element]])
```

Comment:
Should we cache this in ggplot_globals? Building up `theme_gray()` for each element (or those that needs it at least) seems quite resource-needy

Reply:
Yes, good point, will do.